src/clients/c++/perf_analyzer/client_backend/torchserve/http_client.h

```diff
@@ -0,0 +1,164 @@
+///   ...
+///   ...
+/// \endcode
+///
+class HttpClient : public nic::InferenceServerClient {
+ public:
+  ~HttpClient();
+
+  /// Create a client that can be used to communicate with the server.
+  /// \param client Returns a new InferenceServerHttpClient object.
+  /// \param server_url The inference server name and port.
+  /// \param verbose If true generate verbose output when contacting
+  /// the inference server.
+  /// \return Error object indicating success or failure.
+  static Error Create(
+      std::unique_ptr<HttpClient>* client, const std::string& server_url,
+      const bool verbose);
+
+  /// Run synchronous inference on server.
+  /// \param result Returns the result of inference.
+  /// \param options The options for inference request.
+  /// \param inputs The vector of InferInput describing the model inputs.
+  /// \param outputs Optional vector of InferRequestedOutput describing how the
+  /// output must be returned. If not provided then all the outputs in the model
+  /// config will be returned as default settings.
+  /// \param headers Optional map specifying additional HTTP headers to include
+  /// in the metadata of gRPC request.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  Error Infer(
+      InferResult** result, const InferOptions& options,
+      const std::vector<InferInput*>& inputs,
+      const std::vector<const InferRequestedOutput*>& outputs =
+          std::vector<const InferRequestedOutput*>(),
+      const Headers& headers = Headers());
+
+
+ private:
+  HttpClient(const std::string& url, bool verbose);
+  Error PreRunProcessing(
+      void* curl, std::string& request_uri, const InferOptions& options,
+      const std::vector<InferInput*>& inputs,
+      const std::vector<const InferRequestedOutput*>& outputs,
+      const Headers& headers, std::shared_ptr<HttpInferRequest>& request);
+  static size_t InferResponseHeaderHandler(
+      void* contents, size_t size, size_t nmemb, void* userp);
+  static size_t InferResponseHandler(
+      void* contents, size_t size, size_t nmemb, void* userp);
+
+  // The server url
+  const std::string url_;
+  // curl easy handle shared for all synchronous requests
+  void* easy_handle_;
+  curl_mime* mime_handle_;
+};
+
+//======================================================================
+
+class HttpInferRequest {
+ public:
+  HttpInferRequest(
+      TorchServeOnCompleteFn callback = nullptr, const bool verbose = false)
+      : callback_(callback), verbose_(verbose), header_list_(nullptr),
+        response_json_size_(0)
+  {
+  }
+
+  ~HttpInferRequest();
+
+  Error InitializeRequest();
+  nic::RequestTimers& Timer() { return timer_; }
+  std::string& DebugString() { return *infer_response_buffer_; }
+
+  friend HttpClient;
+  friend InferResult;
+
+ private:
+  TorchServeOnCompleteFn callback_;
+  bool verbose_;
+  // Pointer to the list of the HTTP request header, keep it such that it will
```

Comment:
Add extra lines between comment-code pairs for increased readability.

Reply:
I think it will not be needed. These are simple private variables with small description,